### ISSUE TITLE
Pretrained Initializer for different embedding

### ISSUE THREAD

**Issue by Whu-wxy** [allenai/allennlp#2603]
**System (please complete the following information):**
 - Linux
 - Python version: 3.6
 - AllenNLP version: 0.8.3

**Question**

* I want to load a pre-trained model and train with pre-trained word vectors of different dimensions. Is it only necessary to modify "text_field_embedder.tokens.pretrained_file and embedding_dim" in the configuration file, and the projection linear layer initialize by another initializer?

**Comment by joelgrus** [comment]
I'm not sure I understand your question. You want to take a model that's been pre-trained on one set of word vectors and change it to use a different set of word vectors? If you change the embedding dim all of the pretrained weights will have the wrong dimensions.

**Comment by Whu-wxy** [comment]
My idea is to change the following Linear layer,(or add a new linear layer？I don't know which is the general practice.) make it projects new embedding dim to a fix dim.
